types/queries.go

```diff
@@ -90,9 +90,11 @@ type QueryRequest struct {
 }
 
 type BankQuery struct {
-	Supply      *SupplyQuery      `json:"supply,omitempty"`
-	Balance     *BalanceQuery     `json:"balance,omitempty"`
-	AllBalances *AllBalancesQuery `json:"all_balances,omitempty"`
+	Supply           *SupplyQuery           `json:"supply,omitempty"`
+	Balance          *BalanceQuery          `json:"balance,omitempty"`
+	AllBalances      *AllBalancesQuery      `json:"all_balances,omitempty"`
+	DenomMetadata    *DenomMetadataQuery    `json:"get_denom_metadata,omitempty"`
```

Comment:
The JSON must be the snake case of the enum case name `DenomMetadata { denom: String }`, i.e. "denom_metadata" to be compatible

Reply:
Nice catch!